### PR TITLE
submariner: new cask

### PR DESCRIPTION
**Pull request by razvanazamfirei**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [X] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [X] `brew audit --cask --online <cask>` is error-free.
- [X] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [X] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [X] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [X] `brew audit --new-cask <cask>` worked successfully.
- [X] `brew install --cask <cask>` worked successfully.
- [X] `brew uninstall --cask <cask>` worked successfully.

See #146138

### CODE REVIEW

File: Casks/submariner.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  name "Submariner"
+  desc "Subsonic client"
+  homepage "https://submarinerapp.com/"
+
+  app "Submariner.app"
+
+  zap trash: [
+    "~/Library/Application Scripts/fr.read-write.Submariner",
+    "~/Library/Containers/fr.read-write.Submariner",
+    "~/Music/Submariner",
```
Comment (bevanjkay) [inline Casks/submariner.rb:16]: Are we sure that this folder's contents should be trashed?

Comment (razvanazamfirei) [reply Casks/submariner.rb:16]: @cmarhoover should `~/Music/Submariner` be removed under the zap stanza? See below: 
```
zap stanzas should not remove:

Files created by the user directly.
```

Comment (cmarhoover) [reply Casks/submariner.rb:16]: Submariner.app creates the ~/Music/Submariner/ directory automatically on launch. Within this directory, it creates a sqlite.db to store settings, as well as subdirectories for caching album covers and music files for offline listening.

I would want brew to delete this directory when removing the app, but that's just personal preference.

Comment (bevanjkay) [reply Casks/submariner.rb:16]: Sounds like it is suitable to be included in `zap`. Thanks for confirming.

**Review by miccal — APPROVED** [review]

**Pull request merged**